src/hmem_ze.c

```diff
@@ -544,6 +564,41 @@ static void ze_hmem_dl_cleanup(void)
 #endif
 }
 
+static int ze_hmem_find_copy_only_engine(int device_num, int *ordinal)
+{
+	ze_result_t ze_ret;
+	uint32_t cq_grp_count = 0;
+	int i;
+
+	ze_ret = ofi_zeDeviceGetCommandQueueGroupProperties(devices[device_num],
+							    &cq_grp_count, NULL);
+	if (ze_ret)
+		return ze_ret;
+
+	cq_grp_props = calloc(cq_grp_count,
+			      sizeof(ze_command_queue_group_properties_t));
+
+	ze_ret = ofi_zeDeviceGetCommandQueueGroupProperties(devices[device_num],
+							    &cq_grp_count,
+							    cq_grp_props);
+	if (ze_ret)
+		goto err;
+
+	for (i = 0; i < cq_grp_count; i++) {
+		if (cq_grp_props[i].flags &
+		    ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COPY &&
+		    !(cq_grp_props[i].flags &
+		      ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE)) {
+			break;
+		}
+	}
+
+err:
+	free(cq_grp_props);
+	*ordinal = i == cq_grp_count ? 0 : i;
```

Comment:
can you add a note in the commit message that says if no copy only was available it will default to ordinal 0?